app/actions/remote/user.ts

```diff
@@ -332,31 +337,44 @@ export const fetchMissingProfilesByUsernames = async (serverUrl: string, usernam
     }
 };
 
-export const updateAllUsersSinceLastDisconnect = async (serverUrl: string) => {
-    const database = DatabaseManager.serverDatabases[serverUrl];
-    if (!database) {
-        return {error: `${serverUrl} database not found`};
+export const updateAllUsersSince = async (serverUrl: string, since: number, fetchOnly = false) => {
+    if (!since) {
+        return {users: []};
     }
 
-    const lastDisconnectedAt = await queryWebSocketLastDisconnected(database.database);
-
-    if (!lastDisconnectedAt) {
-        return {users: []};
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
     }
-    const currentUserId = await queryCurrentUserId(database.database);
-    const users = await queryAllUsers(database.database);
+
+    const currentUserId = await queryCurrentUserId(operator.database);
+    const users = await queryAllUsers(operator.database);
     const userIds = users.map((u) => u.id).filter((id) => id !== currentUserId);
     let userUpdates: UserProfile[] = [];
     try {
-        userUpdates = await NetworkManager.getClient(serverUrl).getProfilesByIds(userIds, {since: lastDisconnectedAt});
+        userUpdates = await NetworkManager.getClient(serverUrl).getProfilesByIds(userIds, {since});
```

Comment:
The `.getClient` may throw.  Can we call it earlier -  before the queries - in a try/catch?

Reply:
Sure